### PR TITLE
Support traversing the editor navigation history through mouse buttons

### PR DESCRIPTION
**Pull request by martin-fleck-at**
#### What it does
- Adds support for traversing the editor navigation history through mouse buttons 4 and 5, similar to the [VS Code History Service](https://github.dev/microsoft/vscode/blob/c23f0305dbf82b2319b198f4dbf3c5d5bc522f15/src/vs/workbench/services/history/browser/historyService.ts#L114).
- Fixes https://github.com/eclipse-theia/theia/issues/11158

#### How to test
- Open application in browser or electron. If you use the browser you can open your application through a link so you don't have any history to begin with, otherwise the window history navigation will navigate you away from your Theia app. This is also the behavior in the GitHub web editor.
- Navigate within a file or between files.
- Click your mouse button 4 and 5 to navigate between your previous navigations.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/core-preferences.ts [hunk 1]
```diff
@@ -97,6 +97,12 @@ export const corePreferenceSchema: PreferenceSchema = {
             'description': nls.localizeByDefault('Controls whether editors showing a file that was opened during the session should close automatically when getting deleted or renamed by some other process. Disabling this will keep the editor open  on such an event. Note that deleting from within the application will always close the editor and that dirty files will never close to preserve your data.'),
             'default': false
         },
+        'workbench.editor.mouseBackForwardToNavigate': {
+            'type': 'boolean',
+            // nls-todo: Will be available with newer VSCode API
```
Comment (colin-grant-work) [inline packages/core/src/browser/core-preferences.ts:102]: When it's available, this should be caught automatically by the linter (using `localizeByDefault` is an error if the string isn't available; using just `localize` is an error if the exact string is available).

File: packages/editor/src/browser/editor-navigation-contribution.ts [hunk 2]
```diff
@@ -102,6 +103,39 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
...
+            if (change.preferenceName === EditorNavigationContribution.MOUSE_NAVIGATION_PREFERENCE) {
+                updateMouseNavigationListener();
+            }
+        }));
+        updateMouseNavigationListener();
+        this.toDispose.push(mouseNavigationSupport);
+    }
+
+    protected async onMouseDown(event: MouseEvent): Promise<void> {
+        // Support to navigate in history when mouse buttons 4/5 are pressed
```
Comment (colin-grant-work) [inline packages/editor/src/browser/editor-navigation-contribution.ts:128]:
```suggestion
        // Support navigation in history when mouse buttons 4/5 are pressed
```

**Review by colin-grant-work — COMMENTED** [review]
I don't have a mouse that would let me test this as it's meant to function, but some minor comments below.

File: packages/core/src/browser/core-preferences.ts (hunk 1 above)
Comment (martin-fleck-at) [reply packages/core/src/browser/core-preferences.ts:102]: Thank you for the clarification! I noticed that I got the error with `localizeByDefault` and checked what it is you are doing in the `localization-check.js`, it's very nice!

**Review by msujew — APPROVED** [review]
Looks good to me. Although I'm having a bit of trouble getting this to work in the browser (as my browsers try to leave the back when back-navigating), it works perfectly in Electron.

- [x] Back-navigation works correctly using button 3
- [x] Front-navigation works correctly using button 4
- [x] Other mouse buttons continue to work as expected

**Review by JonasHelming — APPROVED** [review]

**Pull request merged**